src/components/BlogList.jsx

```diff
@@ -2,24 +2,49 @@ import BlogPost from "./BlogPost";
 import Pagination from "./Pagination";
 import React from "react";
 import blogs from "../data/blogs.json";
+//-----imported useState.
+import { useState } from "react";
 
 const PAGE_SIZES = [15, 25, 50, 100];
 
 function BlogList() {
-  const currentPaginationData = blogs.posts.slice(0, 15);
+  //-----putting page size into its own state.
+  const [pageSize, setPageSize] = useState(15);
+  //-----putting current page into its own state.
+  const [currentPageNumber, setCurrentPageNumber] = useState(1);
+  //-----declaring start and end of currentPaginationData slice;
+  const startSlice = pageSize * currentPageNumber - pageSize;
+  const endSlice = pageSize * currentPageNumber;
+  //-----changed hard coded slice (0, 15) to variables.
+  const currentPaginationData = blogs.posts.slice(startSlice, endSlice);
+  //-----defining if next and previous buttons are disabled.
+  const isOnlyOnePage = blogs.posts.length <= pageSize;
+  const isLastPage = startSlice <= blogs.posts.length && endSlice >= blogs.posts.length;
+  const isFirstPage = startSlice == 0;
```

Comment:
Strict equality "===" can be used instead of "==" to compare both the values and the types.